src/middleware/cargo_compat.rs

```diff
@@ -1,11 +1,41 @@
-use axum::extract::{MatchedPath, Request};
+use axum::extract::{MatchedPath, Request, State};
 use axum::middleware::Next;
 use axum::response::{IntoResponse, Response};
 use axum::{Extension, Json};
 use http::{header, Method, StatusCode};
+use std::str::FromStr;
+
+#[derive(Clone, Copy, Debug)]
+pub enum StatusCodeConfig {
+    /// Use the original response status code that the backend returned.
+    Disabled,
+    /// Use `200 OK` for all responses to cargo-relevant endpoints.
+    AdjustAll,
+    /// Use `200 OK` for all `2xx` responses to cargo-relevant endpoints, and
+    /// the original status code for all other responses.
+    AdjustSuccess,
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("Failed to parse StatusCodeConfig")]
+pub struct StatusCodeConfigError;
+
+impl FromStr for StatusCodeConfig {
+    type Err = StatusCodeConfigError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "disabled" => Ok(Self::Disabled),
+            "adjust-all" => Ok(Self::AdjustAll),
+            "adjust-success" => Ok(Self::AdjustSuccess),
+            _ => Err(StatusCodeConfigError),
+        }
+    }
+}
```

Comment:
I've changed the implementation from a simple `bool` to an enum. This allows us to use the original status codes in tests, 200 everywhere in production, and e.g. 200 for 2xx on staging or eventually on production too.